README.md
altWinDirStat
=============

An unofficial modification of WinDirStat


WinDirStat/changelog.txt
changelog.txt - What has been done to WinDirStat (newest top)
Last modified: $Date$
=============================================================

2005-07-16	assarbad
	* Release of version 1.1.2 which features new translations and some
	  very minor code corrections.
	
2005-04-17	assarbad
	* Fixed a bug that appeared only in the Spanish, Italian and Czech
	  resource DLLs, but seemed to be by chance. It was the fact that
	  the resource version string which had been moved to the *.rc2
	  files recently, had no explicit statement to which language it
	  belonged. Therefore the Spanish DLL contained the version string
	  as a German string, for example. This confused the ATL code.
	* Changed order of resources manually for one .rc to compare better
	  and track down that error above.
	* Minor changes: added comment and used GetModuleHandle() instead
	  of LoadLibrary() for osspecific.cpp in several places for those
	  DLLs which are anyway already loaded at the time the call is
	  placed.
	* Although included in the respective language DLL the IDS_SEV_* and
	  IDS_FROMsPLATFORMs are now located in the *.rc2 to not translate
	  them by mistake.
	* Changes to the version strings in version.h (resource DLLs need no
	  information whether Unicode or Debug. Development versions show
	  more information than before in the product version.
	* Corrected typo in the Russian translation. Thanks to Dima for
	  pointing it out to me.
	* Credits for translators now in a unicode encoded resource string!
	  Changes to the about dialog have been necessary, obviously.
	  The fact that only a single string with names of translators is
	  being used will greatly simplify maintenance as new translation
	  become available!

2005-04-10	assarbad
	* Added Italian resource DLL project
	* Added Czech resource DLL project
	* Fixed problem with VS-project GUIDs
	* Resource string version now in the *.rc2 files
	* Optimization of resource DLLs saving 4kB each.

2005-02-14	bseifert
	* Added wdsh040e: Hungarian helpfile. Thanks to Nimoy!
	* 2 corrections in Hungarian windirstat.rc.
	* Undone spanish translation of IDS_SEV_CRITICAL, IDS_SEV_GRAVE,
		IDS_SEV_NORMAL, IDS_SEV_WISH and IDS_SEV_FEEDBACK.
		Those must not be translated because they appear in feedback
		mails (we read them!). :-)
	* IDS_LANGUAGERESTARTNOW should be bi-lingual (native language + English).

2005-02-09	assarbad
	* Preparation for Bulgarian translation (without translation
	  yet).

2005-02-08	assarbad
	* Corrections to the Spanish translation after review by Sergio.

2005-02-07	assarbad
	* Adjusted all resources to show Spanish. Missing translation of
	  the word "Spanish" for Hungarian and of the words "Spanish"
	  and "Hungarian" for French.

2005-02-06	assarbad
	* Added Spanish resource dll (thanks to Sergio).

2005-01-29	bseifert
	* Added Hungarian resource dll (thanks to Leonard).
	* Only one build configuration for resource DLLs: UnicodeRelease
	* IDS_ABOUT_AUTHORSTEXTs: obscured e-mail addresses by #-signs to avoid spam

2005-01-02	bseifert
	* Copyright -2005.

2004-12-31	bseifert
	* Added English phrases to IDS_LANGUAGERESTARTNOW (Russian and Polish).
	* Untranslated Polish, French and Russian IDS_SEV_*** and IDS_FROMsPLATFORMs strings.
	* Unicode CString::CompareNoCase doesn't return -1,0,+1 always. Used signum().
	* Bugfix in CMountPoints::IsJunctionPoint().
	* Minor correction configuration.htm.
	* Minor correction in Polish menu.
	
2004-12-30	bseifert
	* setup.exe is no compiled optimizing size (-> ca. -15 KB)
	* Updated testplan.
	* Incremented RESOURCEVERSION to 4.
	
2004-12-25	bseifert
	* Documented Attributes column in helpfiles.
	* Documented Junction points in helpfiles.
	
2004-12-24	bseifert
	* setup: included Polish translation (thanks to Darek).
	* wdsh0415: Polish help file complete! (Thanks to Darek).
	* Help files: switched to a serif font.

2004-12-19	bseifert
	* CMyImageList: bugfix: the background of <Files> etc. was
	  wrong when list had 'stripes' style.
	* Included Sergiy's e-mail address in authors tab.
	* Feedback now routed to feedback-mailing list.
	* Corrected 4 spelling errors in English help file.

2004-12-12	bseifert
	* CMyImageList: improved coloring of <Files>, <Free Space>,
	  and <Unknown> images.
	* CMyImageList: Junction points are now no longer greenified
	  folder icons but a folder icon with a awxlink overlay.
	  
2004-12-12	bseifert
	* Aboutbox: added "Authors" Tab.
	* Copyright now hold by "The authors of WinDirStat".
	* license.txt not needed in localized resource dlls.
	  -> saves 16 KB per resource dll.

2004-11-29	bseifert
	* Introduced SRECT. sizeof(CItem) is now 104 compared to 112.
	* Minor formatting changes.
	
2004-11-28	assarbad
	* Extended CFileFindWDS.
	* Packing of file attributes to keep CItem small.
	* Sorting of file attribute column now implemented.

2004-11-27	bseifert
	* Progress bar now disabled (and Pacman used instead)
	  if "follow junction points" is enabled.
	* The aboutbox and versioninfo now contains a "Unicode"/"Ansi"
	  and a "Debug" specification if applicable.
	* Updated VERSIONINFOs in the windirstat.rc2 files.
	
2004-11-25	assarbad
	* Removal of obsolete functions IsCompressed() and
	  IsEncrypted().
	* Optimised use of GetCompressedFileSize() API which is
	  now cached in the application object similar to PSAPI.
	* Removed manifest and respective file for Russian and
	  the manifest file for German ressources.
	* Fixed coloring bug upon error codes #2 and #123 which
	  happens using the ANSI version on files with Unicode
	  names which cannot be mapped to ANSI names.
	* Implemented the column to show file attributes.

2004-11-24	bseifert
	* Eliminated hidden option "ExplorerStyle". The directory
	  list context menu now appears horizontally at the cursor 
	  position and vertically under (or above) the label.
	  
2004-11-23	bseifert
	* Bugfix: About-Dialog: When the RichEdit control had
	  the focus, pressing the ESC key made id disappear.

WinDirStat/common/version.h
// version.h - Version number. Used by all resource scripts and by aboutdlg.cpp.
//
// WinDirStat - Directory Statistics
// Copyright (C) 2003-2004 Bernhard Seifert
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//
// Author: bseifert@users.sourceforge.net, bseifert@daccord.net
//
// Last modified: $Date$

/*-------------------------------------------------------------------
  This file defines the following:
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  - VN_FILEFLAG_EXE (contains information about RC or debug build)
  - VN_STRING_EXE (contains info about unicode and debug)
  - VN_STRING_DLL (just major.minor.rev.build)
  - VN_FILEVERSION_EXE and VN_PRODVERSION_EXE (only different from
    the DLL version if a development release)
  - VN_FILEVERSION_DLL and VN_PRODVERSION_DLL (numeric representation
    of VN_STRING_DLL)
  - VN_FILEOS_EXE (this depends on Unicode and ANSI!
      For Unicode: VN_FILEOS_EXE == VOS_NT_WINDOWS32
      For ANSI   : VN_FILEOS_EXE == VOS__WINDOWS32 )
  - VN_COPYRIGHTSTRING (copyright to include in the VERSIONINFO)
  - VN_RESOURCEDLL (version of resource DLLs must be the same for DLL
    and EXE at runtime)
  -------------------------------------------------------------------*/

//-------------------------------------------------------------------
// Build categories. Uncomment _one_ line.
//

#define BC_DEVEL			// Development version. The usual setting. File version is 0.0.0.buildno.
//#define BC_RELEASECANDIDATE		// Release candidate. Version number is relevant but not yet official. About-box shows x.y.zrcn. File version is x.y.z.buildno.
//#define BC_RELEASE			// Set this only during official builds. About-box shows x.y.z. File version is x.y.z.buildno

// A release version must not contain debug information. Raise an error!
#if defined(_DEBUG) && defined(BC_RELEASE)
  #error BC_RELEASE _and_ _DEBUG are defined. This must not happen. Releases contain no debug information.
#endif

// This will not change to often, but the years need to be modified
// regularly, so it can be in one central place
#define VN_COPYRIGHTSTRING "Copyright (C) 2003-2005 Bernhard Seifert"

//-------------------------------------------------------------------
// Version number. Relevant for BC_RELEASECANDIDATE and BC_RELEASE.
//
#define VERNUM_MAJOR		1
#define VERNUM_MINOR		1
#define VERNUM_REVISION		2
// The following line is automatically incremented by linkcounter.exe.
// Format: #define blank LINKCOUNT blanks decimal
// Reset this to zero only when you increment VERNUM_MAJOR/MINOR/REVISION.

#define LINKCOUNT  80

//-------------------------------------------------------------------
// Release candidate number. Relevant for BC_RELEASECANDIDATE.
//
#define VERNUM_CANDIDATE	1



/////////////////////////////////////////////////////////////////////
// Derived values from here. Do not edit.

#define VN_BUILD	LINKCOUNT

#define PPSX(s) #s
#define PPS(s) PPSX(s)

#ifdef _UNICODE
	#define UASPEC "Unicode"
	// OS version is only relevant for the EXE
	#define VN_FILEOS_EXE VOS_NT_WINDOWS32
#else
	#define UASPEC "Ansi"
	// OS version is only relevant for the EXE
	#define VN_FILEOS_EXE VOS__WINDOWS32
#endif

#ifdef _DEBUG
	#define DRSPEC " Debug"
#else
	#define DRSPEC ""
#endif

#define VERVARIANT " (" UASPEC DRSPEC ")"

// This is just major.minor.rev.build always!
#define VN_STRING_DLL	PPS(VERNUM_MAJOR) "." PPS(VERNUM_MINOR) "." PPS(VERNUM_REVISION) "." PPS(VN_BUILD)

#if defined(BC_DEVEL)

	#define VN_MAJOR	0
	#define VN_MINOR	0
	#define VN_REVISION	0
	#define VN_FILEFLAG	0
	#define VN_STRING_DLL	PPS(VERNUM_MAJOR) "." PPS(VERNUM_MINOR) "." PPS(VERNUM_REVISION) "." PPS(VN_BUILD)
	// The variant (debug or not/ unicode or not) is not relevant for resource DLLs, but for EXEs
	#define VN_STRING_EXE	VN_STRING_DLL " devel" VERVARIANT

#elif defined(BC_RELEASECANDIDATE)

	#define VN_MAJOR	VERNUM_MAJOR
	#define VN_MINOR	VERNUM_MINOR
	#define VN_REVISION	VERNUM_REVISION
	#define VN_FILEFLAG	VS_FF_PRERELEASE
	// The variant (debug or not/ unicode or not) is not relevant for resource DLLs, but for EXEs
	#define VN_STRING_EXE	VN_STRING_DLL "rc" PPS(VERNUM_CANDIDATE) VERVARIANT

#elif defined(BC_RELEASE)

	#define VN_MAJOR	VERNUM_MAJOR
	#define VN_MINOR	VERNUM_MINOR
	#define VN_REVISION	VERNUM_REVISION
	#define VN_FILEFLAG	0
	// The variant (debug or not/ unicode or not) is not relevant for resource DLLs, but for EXEs
	#define VN_STRING_EXE	VN_STRING_DLL VERVARIANT

#endif

// EXE files have a different version number in development releases
#define VN_FILEVERSION_EXE VN_MAJOR,VN_MINOR,VN_REVISION,VN_BUILD
#define VN_PRODVERSION_EXE VN_FILEVERSION_EXE
// Resource DLLs need no different version number
#define VN_FILEVERSION_DLL VERNUM_MAJOR,VERNUM_MINOR,VERNUM_REVISION,VN_BUILD
#define VN_PRODVERSION_DLL VN_FILEVERSION_DLL

// Whether debug or not is not relevant for resource DLLs
#ifdef _DEBUG
 #define VN_FILEFLAG_EXE VS_FF_DEBUG | VN_FILEFLAG
#else
 #define VN_FILEFLAG_EXE VN_FILEFLAG
#endif

// ...nothing else.
#undef BC_DEVEL
#undef BC_RELEASECANDIDATE
#undef BC_RELEASE


// $Log$
// Revision 1.38  2005/07/16 13:58:05  assarbad
// *** empty log message ***
//
// Revision 1.37  2005/07/16 13:38:44  assarbad
// - Preparation for release 1.1.2
//
// Revision 1.36  2005/04/17 20:45:15  assarbad
// - Now the list of translators is shared among all translations. See changelog for details.
//
// Revision 1.35  2005/04/17 18:13:42  assarbad
// - Moved some "static" resource strings into the respective *.rc2 files
// - Corrected typo in Russian DLL
// - Modified behavior of VERSIONINFO for DLLs. "version.h" has changed therefore
// ... for details as usual, see the changelog.
//
// Revision 1.34  2005/04/17 12:27:17  assarbad
// - For details see changelog of 2005-04-17
//
// Revision 1.33  2005/04/10 16:49:22  assarbad
// - Some smaller fixes including moving the resource string version into the rc2 files
//
// Revision 1.32  2005/04/10 14:57:09  assarbad
// - Added Italian and Czech. See changelog for details.
//
// Revision 1.31  2005/02/14 23:03:22  bseifert
// Add Hungarian help file. Minor resource file corrections.
//
// Revision 1.30  2005/02/07 16:22:28  assarbad
// - Adjusted all resources to show Spanish. Missing translation of the word
//  "Spanish" for Hungarian and of the words "Spanish" and "Hungarian" for French.
//
// Revision 1.29  2005/01/29 07:08:13  bseifert
// Added Hungarian resource dll.
//
// Revision 1.28  2005/01/02 03:56:04  bseifert
// Copyright -2005. Release 1.1.1
//
// Revision 1.27  2004/12/31 16:01:38  bseifert
// Bugfixes. See changelog 2004-12-31.
//
// Revision 1.26  2004/12/30 11:26:12  bseifert
// Decided to use serif font for help files. Incremented version number. rc1.
//
// Revision 1.25  2004/12/30 10:18:39  bseifert
// Updated RESOURCEVERSION. Updated testplan. setup: "minimize size".
//
// Revision 1.24  2004/12/30 08:01:11  bseifert
// helpfile updated.
//
// Revision 1.23  2004/12/25 13:41:46  bseifert
// Synced help files.
//
// Revision 1.22  2004/12/24 10:39:27  bseifert
// Added Polish setup and  Polish help file.
//
// Revision 1.21  2004/12/19 10:52:36  bseifert
// Minor fixes.
//
// Revision 1.20  2004/12/12 13:40:47  bseifert
// Improved image coloring. Junction point image now with awxlink overlay.
//
// Revision 1.19  2004/12/12 08:34:56  bseifert
// Aboutbox: added Authors-Tab. Removed license.txt from resource dlls (saves 16 KB per dll).
//
// Revision 1.18  2004/11/27 07:19:36  bseifert
// Unicode/Ansi/Debug-specification in version.h/Aboutbox. Minor fixes.
//
// Revision 1.17  2004/11/25 23:07:22  assarbad
// - Derived CFileFindWDS from CFileFind to correct a problem of the ANSI version
//
// Revision 1.16  2004/11/13 18:48:29  bseifert
// Few corrections in Polish windirstat.rc. Thanks to Darek.
//
// Revision 1.15  2004/11/13 08:17:04  bseifert
// Remove blanks in Unicode Configuration names.
//
// Revision 1.14  2004/11/12 21:03:53  bseifert
// Added wdsr0415.dll. New output dirs for Unicode. Minor corrections.
//
// Revision 1.13  2004/11/12 00:47:42  assarbad
// - Fixed the code for coloring of compressed/encrypted items. Now the coloring spans the full row!
//
// Revision 1.12  2004/11/09 23:23:03  bseifert
// Committed new LINKCOUNT to demonstrate merge conflict.
//
// Revision 1.11  2004/11/07 20:14:30  assarbad
// - Added wrapper for GetCompressedFileSize() so that by default the compressed file size will be shown.
//
// Revision 1.10  2004/11/07 00:51:30  assarbad
// *** empty log message ***
//
// Revision 1.9  2004/11/07 00:06:34  assarbad
// - Fixed minor bug with ampersand (details in changelog.txt)
//
// Revision 1.8  2004/11/05 16:53:05  assarbad
// Added Date and History tag where appropriate.
//
